src/back/spv/instructions.rs

```diff
@@ -478,6 +478,34 @@ pub(super) fn instruction_composite_construct(
 // Arithmetic Instructions
 //
 
+pub(super) fn instruction_i_mul(
```

Comment:
would you consider generalizing it as `instruction_binary_op()` that receives and `Op` and can be used for all of the binary operators as well as multiplications, to save code?
If you see the clear names like `instruction_matrix_times_scalar` bringing value, you can still have them as thin layers on top of `instruction_binary_op`

Reply:
Yes I understand what you are suggesting, but I rather would not have the developer of the SPIR-V back-end use Op codes directly, that is the reason for this instructions module.

What we can do is still creating the `instruction_binary_op`, but make it private, and exposing the instructions like `instruction_matrix_times_scalar`, but in their body, they will call `instruction_binary_op` with the op as argument.

Reply:
Update: I see not all binary instructions follow the layout, and will diverge. Can I suggest to keep it like this, and when we have all binary instructions implemented, we can clearly see where "duplicate" is?

Reply:
sure thing